src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -2661,7 +2667,7 @@ protected function generateTree($path, $intMargin, $mount=false, $blnProtected=t
 			}
 		}
 
-		if ($this->blnHideFiles)
+		if (!$this->blnFiles)
```

Comment:
This is wrong and should be `if (!$this->blnFiles && ! $this->blnFilesOnly)`. See:

https://github.com/contao/core-bundle/blob/master/src/Resources/contao/widgets/FileSelector.php#L535